### DISCUSSION TITLE
Overwrite reset password link in email templates

### DISCUSSION THREAD

**Discussion by valerensi** [directus/directus#5728] [Archive]
Hello,
I have to make some change on default reset password template. 
I found out some discussion about the chance to use custom template, but I'm not sure if the feature has been added to my version (9.0.0-rc.55) and how it works. 

I also join to Discord channel but I found out only messages that suggest to overwrite the default template and is not a brilliant solution because it will be replaced with every core updates. 

Anyone can help me? 
Thanks :)

**Comment by rijkvanzanten** [comment]
Heya!

You can set the reset password link in the request: https://docs.directus.io/reference/api/system/authentication/#request-password-reset (it has to be whitelisted in [`PASSWORD_RESET_URL_ALLOW_LIST`](https://docs.directus.io/reference/environment-variables/#security). 

> that suggest to overwrite the default template and is not a brilliant solution because it will be replaced with every core updates.

This is not true. The custom templates are in the extensions folder, and aren't tracked in the Directus releases. They're an extension type 👍🏻 

> but I'm not sure if the feature has been added to my version (9.0.0-rc.55) 

I would highly recommend updating to the latest version, as I'm not 100% sure what parts of my answer were available in 55 👍🏻

**Reply by valerensi** [reply]
I have tried to add a custom template email in extensions folder but it seams it doesn't work. For this reason I'm asking if anyone can help me with some example or more information.

**Reply by rijkvanzanten** [reply]
@valerensi Make sure to update your instance first, custom template overrides definitely landed more recently than 55 👍🏻

**Reply by valerensi** [reply]
![E16DD616-F4C7-4A55-8033-94F8E2BC7320](https://user-images.githubusercontent.com/5748709/118527828-f40a3900-b741-11eb-843e-9df548bf543b.png)
![2976065F-AB5D-4C5B-B08F-850A2254D25B](https://user-images.githubusercontent.com/5748709/118527831-f53b6600-b741-11eb-8e87-9b4944c37a15.png)

Of course I will update my instance. Can you please explain how I have to proceed to overwrite the email template?
If I have correctly understand I have to add a folder called “templates” inside “extensions”.
Than what do I have to do? I need to copy inside all dist -> mail folder or only dist -> mail -> templates one?
Thanks for your help :) and sorry for too many questions.

**Reply by rijkvanzanten** [reply]
> If I have correctly understand I have to add a folder called “templates” inside “extensions”.

That's correct! Then in there you can add custom ones with any name (to use when sending custom emails). If you create a custom template in there with the same name as a system one (eg `password-reset.liquid` or `user-invitation.liquid`) that's used instead of the system one. You don't have to copy paste them for them to work (though you'll most likely want to keep the `{{ url }}` in there somewhere to render the link to the reset page 🙂

**Reply by valerensi** [reply]
Ok so the structure will be extension -> templates -> password-reset.liquid wright?

**Reply by rijkvanzanten** [reply]
Yup!

**Reply by valerensi** [reply]
Thank you very much for your help!

**Reply by benhaynes** [reply]
Ref: https://github.com/directus/directus/issues/5736
